test/over_react_test/react_util_test.dart

```diff
@@ -744,39 +745,87 @@ main() {
       });
     });
 
-    group('queryByTestId returns the topmost Element that has the appropriate value for the', () {
-      group('`data-test-id` html attribute key', () {
-        test('', () {
-          var renderedInstance = render((Nested()..addTestId('value'))());
+    group('queryByTestId returns the topmost Element', () {
+      group('that has the appropriate value for the', () {
+        group('`data-test-id` html attribute key', () {
+          test('', () {
+            var renderedInstance = render((Nested()..addTestId('value'))());
+            var innerNode = findDomNode(renderedInstance).querySelector('[data-test-id~="inner"]');
+
+            expect(queryByTestId(renderedInstance, 'value'), innerNode);
+          });
+
+          test('expect when no matching element exists', () {
+            var renderedInstance = render(Dom.div()());
+
+            expect(queryByTestId(renderedInstance, 'value'), isNull);
+          });
+        });
+
+        test('custom html attribute key', () {
+          var renderedInstance = render((Nested()..addTestId('value', key: 'data-custom-id'))());
           var innerNode = findDomNode(renderedInstance).querySelector('[data-test-id~="inner"]');
 
-          expect(queryByTestId(renderedInstance, 'value'), innerNode);
+          expect(queryByTestId(renderedInstance, 'value', key: 'data-custom-id'), innerNode);
         });
 
-        test('expect when no matching element exists', () {
+        test('`data-test-id` html attribute, expect when no matching element exists', () {
           var renderedInstance = render(Dom.div()());
 
           expect(queryByTestId(renderedInstance, 'value'), isNull);
         });
       });
 
-      test('custom html attribute key', () {
-        var renderedInstance = render((Nested()..addTestId('value', key: 'data-custom-id'))());
-        var innerNode = findDomNode(renderedInstance).querySelector('[data-test-id~="inner"]');
+      group('from within a ShadowRoot when `searchInShadowDom` is `true`', () {
+        test('', () async {
+          var jacket = mount(ShadowNested()());
+
+          // Let the shadow dom mount (the test components kinda slow since it does it after adding it to the dom.)
+          await Future.delayed(const Duration(milliseconds: 500));
```

Comment:
Instead of a fixed delay (this one is pretty large), you can use `await pumpEventQueue()` instead

Reply:
```suggestion
          await pumpEventQueue();
```